eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -256,6 +256,49 @@ public function testCreateLocationThrowsInvalidArgumentExceptionRemoteIdExists()
         /* END: Use Case */
     }
 
+    /**
+     * Test for the createLocation() method.
+     *
+     * @see \eZ\Publish\API\Repository\LocationService::createLocation()
+     * @depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testNewLocationCreateStruct
+     * @dataProvider dataProviderForOutOfRangeLocationPriority
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function testCreateLocationThrowsInvalidArgumentExceptionPriorityIsOutOfRange($priority)
+    {
+        $repository = $this->getRepository();
+
+        $contentId = $this->generateId('object', 41);
+        $parentLocationId = $this->generateId('location', 5);
+        /* BEGIN: Use Case */
+        // $contentId is the ID of an existing content object
+        // $parentLocationId is the ID of an existing location
+        $contentService = $repository->getContentService();
+        $locationService = $repository->getLocationService();
+
+        // ContentInfo for "How to use eZ Publish"
+        $contentInfo = $contentService->loadContentInfo($contentId);
+
+        $locationCreate = $locationService->newLocationCreateStruct($parentLocationId);
+        $locationCreate->priority = $priority;
+        $locationCreate->hidden = true;
+        $locationCreate->remoteId = 'sindelfingen';
+        $locationCreate->sortField = Location::SORT_FIELD_NODE_ID;
+        $locationCreate->sortOrder = Location::SORT_ORDER_DESC;
+
+        // Throws exception, since priority is out of range
+        $locationService->createLocation(
+            $contentInfo,
+            $locationCreate
+        );
+        /* END: Use Case */
+    }
+
+    public function dataProviderForOutOfRangeLocationPriority()
+    {
+        return [[-2147483649], [2147483648]];
```

Comment:
How about `DomainMapper::MIN_LOCATION_PRIORITY - 1`  and `DomainMapper::MAX_LOCATION_PRIORITY + 1`?

Reply:
(skipped this as integration test should not know about internal core details)

Reply:
sorry to overrule your suggestion @alongosz but way to much core stuff is creeping into integration tests lately so lets try to avoid it as much as possible going forward on anything but SetupFactory et al.